### PR TITLE
Allow snapping photos/videos from home screen

### PR DESCRIPTION
**Pull request by mikeger**
- Uses iOS integrated camera UI.
- Uses forwarding UI.

### CODE REVIEW

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -618,6 +618,9 @@
 		87D5E4401C3E797A00EB8289 /* UserClientFingerprintTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 87D5E43F1C3E797A00EB8289 /* UserClientFingerprintTests.swift */; };
 		87D5E4421C3E79C600EB8289 /* NSData+Fingerprint.swift in Sources */ = {isa = PBXBuildFile; fileRef = 87D5E4411C3E79C600EB8289 /* NSData+Fingerprint.swift */; };
 		87D5E4431C3E79DA00EB8289 /* UserClient+Fingerprint.swift in Sources */ = {isa = PBXBuildFile; fileRef = BF86E11A1C245BF9001D9430 /* UserClient+Fingerprint.swift */; };
+		87D679E41EBB72E8005BEB4C /* CameraPicker.swift in Sources */ = {isa = PBXBuildFile; fileRef = 87D679E31EBB72E8005BEB4C /* CameraPicker.swift */; };
+		87D679E61EBC8002005BEB4C /* String.swift in Sources */ = {isa = PBXBuildFile; fileRef = 87D679E51EBC8002005BEB4C /* String.swift */; };
```
Comment (daehn) [inline Wire-iOS.xcodeproj/project.pbxproj]: `String.swift` sounds a bit misleading I think, maybe `String+Helper.swift` might be more suitable.

File: Wire-iOS/Sources/UserInterface/ConversationList/ConversationListViewController+CameraPicker.swift [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            }
+        }
+        
+        cameraPicker.pick()
+    }
+    
+    public func showShareControllerFor(image: UIImage) {
+        let conversations = SessionObjectCache.shared().allConversations.shareableConversations()
+        
+        let shareViewController: ShareViewController<ZMConversation, UIImage> = ShareViewController(shareable: image, destinations: conversations)
```
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/ConversationList/ConversationListViewController+CameraPicker.swift]: We wrap the `ShareViewController` in a `KeyboardAvoidingViewController` in the other places were we present it, would be good to do the same here.

File: Wire-iOS/Sources/UserInterface/ConversationList/ConversationListViewController+CameraPicker.swift [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+            self.dismiss(animated: true)
+        }
+        
+        self.present(shareViewController, animated: true)
+    }
+    
+    public func showShareControllerFor(videoAtURL: URL) {
+        let conversations = SessionObjectCache.shared().allConversations.shareableConversations()
+        
+        let shareViewController: ShareViewController<ZMConversation, URL> = ShareViewController(shareable: videoAtURL, destinations: conversations)
```
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/ConversationList/ConversationListViewController+CameraPicker.swift]: Same here.

File: Wire-iOS/Sources/UserInterface/ConversationList/CameraPicker.swift [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        pickerController.allowsEditing = false
+        pickerController.mediaTypes = [kUTTypeImage as String, kUTTypeMovie as String]
+        pickerController.videoMaximumDuration = ConversationUploadMaxVideoDuration
+        pickerController.transitioningDelegate = FastTransitioningDelegate.sharedDelegate
+        
+        if sourceType == .camera {
+            switch Settings.shared().preferredCamera {
+            case .back:
+                pickerController.cameraDevice = .rear
+                break
```
Comment (vytis) [inline Wire-iOS/Sources/UserInterface/ConversationList/CameraPicker.swift]: Not really needed here and in the other `case`

File: Wire-iOS/Sources/UserInterface/ConversationList/ConversationListViewController+CameraPicker.swift [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    }
+}
+
+
+extension ConversationListViewController {
+
+    @objc public func showCameraPicker() {
+        let cameraPicker = CameraPicker(target: self)
+        cameraPicker.didPickResult = { result in
+            switch result {
```
Comment (vytis) [inline Wire-iOS/Sources/UserInterface/ConversationList/ConversationListViewController+CameraPicker.swift]: Make self weak here maybe?

**Comment by daehn** [comment]
@mikeger Some of the `ConversationListBottomBarTests` failed on Circle

**Comment by mikeger** [comment]
Thanks for the comments, indeed tests are broken, since the new button is added.

**Comment by mikeger** [comment]
@vytis @daehn updated!

**Pull request merged**